### PR TITLE
Makes mech don't suck while destroying walls

### LINKED ISSUE #15128: Mech hit RNG
Mechs are kind of shit as an option for OPs in particular and the main reason for this is their inability to get around.

Fire doors completely stop them in their tracks, and it can literally take 40 hits per table or wall to break it when a fucking wispy hivelord can knock a table out in one hit. Mechs should at least be able to smash tables and windows in one hit and ideally walls too. I've spent multiple mech ops rounds stuck in one patch of hallway grinding at a table that just won't break because I'm unlucky and the odds are awful.

Please fix. Tables, windows and walls should really be smashable in one hit by the big mechs

### PR DESCRIPTION
**Pull request by lzimann**
Fixes #15128
Also makes them destroy table in one hit. 
They now use the hardness var as a chance of destroying a wall, same chance as a hulk has to destroy. 
:cl: Lzimann
tweak: Combat Mechas now have a increased chance of destroying a wall with punches (40% for walls and 20% for reinforced walls)!
rscadd: Combat mechas can now destroy tables and racks with a punch!
/:cl:

### CODE REVIEW

**Comment by AnturK** [comment]
Probably would be better if the mechs destroyed the tables on movement, so non-combat mechs can also trample them. Also racks.

**Comment by Lumipharon** [comment]
This will let mechs slice down r-walls like butter, if it's like hulk. Time for maulers to require zero support from their team I suppose.

**Comment by Buggy123** [comment]
IMO they could probably use a buff, I haven't seen nuke ops actually use one in ages.

**Comment by Lumipharon** [comment]
That's because gygax has shitty fire bullets, so aren't actually great at killing things, and the mauler costs fuckloads of points, which means the other ops don't get much kit. 
Atleast with how it stands, those other ops are still useful for emagging shit/helping the maulers mobility. After this, they're just glorified repair drones.

**Comment by lzimann** [comment]
That doesn't change the fact mechs are still weak and even after this PR they won't be used that much

**Comment by Zilenan91** [comment]
We can't buff mech damage/survivability. If we do, xenos will get pushed a few feet further into the dumpster.

**Comment by Lumipharon** [comment]
>mauler
>weak

If you die in a mauler you're either a terrible pilot or you didn't get rid of the ion rifle

**Comment by lzimann** [comment]
Nobody uses mechs anyway, hopefully this will make they a little more viable

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -551,7 +551,8 @@
 					step(obstacle, dir)
 			else if(istype(obstacle, /mob))
 				step(obstacle, dir)
-
+	if(istype(obstacle, /obj/structure/table) || istype(obstacle, /obj/structure/rack))
```
Comment (Cheridan) [inline code/game/mecha/mecha.dm]: this is kinda bad. also unnecessary? just knock it over with a click

Comment (lzimann) [inline code/game/mecha/mecha.dm]: Hm, it was suggested by Kaze and I liked it, what's so bad about it?

Comment (tkdrg) [inline code/game/mecha/mecha.dm]: probably refering to putting istype()s on Bump() which is snowflake, specially when you're already handling it on melee attack

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -45,15 +45,28 @@
...
-		if(prob(25))
-			table_destroy(1)
+	switch(severity)
+		if(1)
+			qdel(src)
+		if(2)
+			if(prob(50))
+				table_destroy(1)
+			else
+				qdel(src)
```
Comment (tkdrg) [inline code/game/objects/structures/tables_racks.dm]: Severities 1 & 2 are already handled in the parent. Don't call parent if you're gonna do this

**Comment by Lumipharon** [comment]
whatevs.

**Pull request merged**